autogpt_platform/backend/backend/api/features/credits_rate_limit.py

```
@@ -0,0 +1,79 @@
+    Redis blip must never block the billing status read for every user, which
+    would be far worse than one client briefly exceeding its cap.
+    """
+    now = datetime.now(UTC)
+    key = _window_key(user_id, now=now)
+    try:
+        redis = await get_redis_async()
+        # Atomic create-with-TTL, then INCR. SET NX EX makes the TTL part of
+        # the same write that creates the key; later INCRs preserve it.
+        await redis.set(key, 0, ex=SUBSCRIPTION_STATUS_WINDOW_SECONDS, nx=True)
+        count = await redis.incr(key)
+    except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)